cpi/c/src/main.c

```diff
@@ -0,0 +1,52 @@
+/**
+ * @brief C-based CPI BPF program
+ */
+#include <solana_sdk.h>
+
+/// Amount of bytes of account data to allocate
+#define SIZE 42
+
+extern uint64_t entrypoint(const uint8_t *input)
+{
+  SolAccountInfo accounts[2];
+  SolParameters params = (SolParameters){.ka = accounts};
+
+  if (!sol_deserialize(input, &params, SOL_ARRAY_SIZE(accounts)))
+  {
+    return ERROR_INVALID_ARGUMENT;
+  }
+
+  SolAccountInfo allocated_info = params.ka[0];
+  SolAccountInfo system_program_info = params.ka[1];
+
+  uint8_t seed[] = {'Y', 'o', 'u', ' ', 'p', 'a', 's', 's',
+                    ' ', 'b', 'u', 't', 't', 'e', 'r'};
+  const SolSignerSeed seeds[] = {{seed, SOL_ARRAY_SIZE(seed)},
+                                 {&params.data[0], 1}};
+
+  const SolSignerSeeds signers_seeds[] = {{seeds, SOL_ARRAY_SIZE(seeds)}};
+
+  SolPubkey expected_allocated_key;
+  if (SUCCESS != sol_create_program_address(seeds, SOL_ARRAY_SIZE(seeds),
+                                            params.program_id,
+                                            &expected_allocated_key))
+  {
+    return ERROR_INVALID_INSTRUCTION_DATA;
+  }
+
+  if (!SolPubkey_same(&expected_allocated_key, allocated_info.key))
+  {
+    return ERROR_INVALID_ARGUMENT;
+  }
+
+  SolAccountMeta arguments[] = {{allocated_info.key, true, true}};
+  uint8_t data[4 + 8];
+  *(uint16_t *)data = 8;
+  *(uint64_t *)(data + 4) = SIZE;
```

Comment:
nit: Can you add the comments from the original here?
```suggestion
  uint8_t data[4 + 8];            // Enough room for the Allocate instruction
  *(uint16_t *)data = 8;          // Allocate instruction enum value
  *(uint64_t *)(data + 4) = SIZE; // Size to allocate
```

Reply:
added 7559567000a6b07185f3dec78b023b89b4d4400e